### PR TITLE
test: refactor chmod tests to AVA

### PR DESCRIPTION
**Pull request by nfischer**
This is an automated PR, so double check that I got all the test cases. I'll follow up once I've verified that this translates everything correctly.

### CODE REVIEW

File: ava-test/chmod.js [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+  t.is(result.code, 0);
+  result = shell.chmod('744', `${t.context.tmp}/chmod/xdir/deep`);
+  t.is(result.code, 0);
+  result = shell.chmod('644', `${t.context.tmp}/chmod/xdir/deep/file`);
+  t.is(result.code, 0);
+  result = shell.chmod('-R', 'a+X', `${t.context.tmp}/chmod/xdir`);
+  t.is(result.code, 0);
+});
+
+test('No Test Title #27', t => {
```
Comment (freitagbr) [inline ava-test/chmod.js]: We need a title for this test, though I'm not sure what it's testing.

File: ava-test/chmod.js [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+  let result = shell.chmod('blah');
+  t.truthy(shell.error());
+  t.is(result.code, 1);
+  result = shell.chmod('893', `${t.context.tmp}/chmod`);  // invalid permissions - mode must be in octal
+  t.truthy(shell.error());
+  t.is(result.code, 1);
+});
+
+// TODO(nate): fix this description
+test('Test files - the bitmasking is to ignore the upper bits.', t => {
```
Comment (freitagbr) [inline ava-test/chmod.js]: Fix this description too?

File: ava-test/chmod.js (hunk 1 above)
Comment (nfischer) [inline ava-test/chmod.js]: I think this was actually a cleanup step before. There's no harm in running these lines and making sure they succeed though, so I kept them.

File: ava-test/chmod.js (hunk 2 above)
Comment (nfischer) [inline ava-test/chmod.js]: done

**Comment by nfischer** [comment]
@freitagbr PTAL.

 - I did the changes requested
 - I made `BITMASK` a constant
 - I changed this to be a `before()` instead of a `beforeEach()` since all these unit tests must run serially (they depend on previous tests). We should probably go through and actually fix this, though.

**Comment by freitagbr** [comment]
LGTM

**Comment by nfischer** [comment]
LGTM

**Pull request merged**